lumicks/pylake/correlated_stack.py

```diff
@@ -487,15 +487,26 @@ def stop(self):
     @deprecated(
         reason=(
             "For camera based images only the integration start/stop timestamps are defined. "
-            "Use `CorrelatedStack.frame_timestamp_ranges` instead."
+            "Use `CorrelatedStack.frame_timestamp_ranges()` instead."
         ),
         action="always",
         version="0.11.1",
     )
     def timestamps(self):
-        return self.frame_timestamp_ranges
+        return self.frame_timestamp_ranges()
 
-    @property
-    def frame_timestamp_ranges(self):
-        """List of time stamps."""
-        return [frame.frame_timestamp_range for frame in self]
+    def frame_timestamp_ranges(self, exclude=True):
+        """Get start and stop timestamp of each frame in the stack.
+
+        Parameters
+        ----------
+        exclude : bool
+            Exclude dead time at the end of each frame.
+        """
+        ts_ranges = [frame.frame_timestamp_range for frame in self]
+        if exclude:
+            return ts_ranges
+        else:
+            frame_ts = [(start[0], stop[0]) for start, stop in zip(ts_ranges, ts_ranges[1:])]
+            frame_ts.append((ts_ranges[-1][0], ts_ranges[-1][1]))
```

Comment:
Can the frame rate of the `CorrelatedStacks` be assumed constant? If so one could calculate dt and add the last timestamp range like:
```suggestion
try:
    if len(ts_ranges) > 2:
        dt = ts_ranges[-1][0] - ts_ranges[-2][0]
        stop = ts_ranges[-1][0] + dt
    else:  # len <= 1
        stop = ts_ranges[-1][1]
    frame_ts.append((ts_ranges[-1][0], stop))
except IndexError:
    pass
```
This would also allow for empty timestamp ranges.

Reply:
```suggestion
try:
    if len(ts_ranges) >= 2:
        dt = ts_ranges[-1][0] - ts_ranges[-2][0]
        stop = ts_ranges[-1][0] + dt
    else:  # len <= 1
        stop = ts_ranges[-1][1]
    frame_ts.append((ts_ranges[-1][0], stop))
except IndexError:
    pass
```

Reply:
It can't be assumed constant. You can change the camera framerate while acquiring. One can not even assume that there is more than one frame since the stack may be a snapshot.

Reply:
I thought it might be useful to also allow for empty stacks and forgot that we do not allow for empty `CorrelatedStacks` when we slice.